lambdas/question/src/main/java/uk.gov.di.ipv.cri.kbv.api.handler/QuestionHandler.java

```diff
@@ -43,83 +46,83 @@ public QuestionHandler() {
     }
 
     @Override
     @Tracing(captureMode = CaptureMode.DISABLED)
     public APIGatewayProxyResponseEvent handleRequest(
             APIGatewayProxyRequestEvent input, Context context) {
-
         String responseBody = "{}";
         Map<String, String> responseHeaders = Map.of("Content-Type", "application/json");
         String json;
         int statusCode;
 
         try {
             String sessionId = input.getHeaders().get(HEADER_SESSION_ID);
-            KBVSessionItem kbvSessionItem = storageService.getSessionId(sessionId);
+            KBVSessionItem kbvSessionItem =
+                    storageService.getSessionId(sessionId).orElseThrow(NullPointerException::new);
+            PersonIdentity personIdentity =
+                    objectMapper.readValue(
+                            kbvSessionItem.getUserAttributes(), PersonIdentity.class);
+
             QuestionState questionState =
                     objectMapper.readValue(kbvSessionItem.getQuestionState(), QuestionState.class);
-            PersonIdentity personIdentity = questionState.getPersonIdentity();
             json = objectMapper.writeValueAsString(personIdentity);
             Optional<Question> nextQuestion = questionState.getNextQuestion();
             if (nextQuestion.isEmpty()) { // we should fall in this block once only
                 // fetch a batch of questions from experian kbv wrapper
-                QuestionsResponse questionsResponse = startGetQuestions(json);
-                if (!questionState.setQuestionsResponse(questionsResponse)) {
-                    statusCode = 400;
+                QuestionsResponse questionsResponse = experianService.getQuestions(json);
+                if (questionState != null
+                        && !questionState.setQuestionsResponse(questionsResponse)) {
+                    statusCode = HttpStatus.SC_BAD_REQUEST;
                     responseBody = "{ \"error\":\" no further questions \" }";
                 } else {
-                    statusCode = 200;
+                    statusCode = HttpStatus.SC_OK;
                     String state = objectMapper.writeValueAsString(questionState);
-                    storageService.save(sessionId, state);
+                    storageService.update(
+                            sessionId,
+                            state,
+                            questionState.getControl().getAuthRefNo(),
+                            questionState.getControl().getURN());
                     nextQuestion = questionState.getNextQuestion();
                     responseBody = objectMapper.writeValueAsString(nextQuestion.get());
                 }
             } else {
                 // TODO Handle scenario when no questions are available
-                statusCode = 200;
+                statusCode = HttpStatus.SC_OK;
+                nextQuestion = questionState.getNextQuestion();
+                if (!nextQuestion.isEmpty()) {
```

Comment:
nextQuestion.isPresent

Reply:
Changed handler and handler test code to implement the same